src/iop/highlights.c

```diff
@@ -2030,6 +2031,8 @@ void commit_params(struct dt_iop_module_t *self, dt_iop_params_t *p1, dt_dev_pix
   dt_iop_highlights_data_t *d = (dt_iop_highlights_data_t *)piece->data;
 
   memcpy(d, p, sizeof(*p));
+  if(d->mode > DT_IOP_HIGHLIGHTS_MAXMODE)
```

Comment:
don't we do that automatically when loading module data using the introspection circuitry based on the description in `dt_iop_highlights_params_t`?

Reply:
I did think so too but as the `dt_iop_highlights_mode_t` field gets extended in my branch with enum = 5 or 6 i ran into issues so had to click the reset to default button.

In don't understand the introspection stuff well enough but it would surprise me that it checks for defined enum data, at least it didn't work for me.

Reply:
But anyway,  think this is not a good approach. First it will impact only the devs and second we have such possible failure in many many places. So the proper fix would be to have the introspection circuitry fixed or enhanced to take this into account.

Reply:
ok, no problem and after rethinking the issue you are fully correct.